azure-devops/azext_devops/dev/team/security_group.py

```diff
@@ -0,0 +1,202 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from __future__ import print_function
+from knack.log import get_logger
+from knack.util import CLIError
+from azext_devops.devops_sdk.exceptions import AzureDevOpsClientRequestError
+from azext_devops.devops_sdk.v5_0.graph.models import (JsonPatchOperation,
+                                                       GraphSubjectLookup,
+                                                       GraphSubjectLookupKey)
+from azext_devops.dev.common.identities import resolve_identity_as_id
+from azext_devops.dev.common.services import (get_graph_client,
+                                              get_project_id_from_name,
+                                              resolve_instance)
+from .security_group_helper import (GraphGroupVstsCreationContext,
+                                    GraphGroupMailAddressCreationContext,
+                                    GraphGroupOriginIdCreationContext)
+
+logger = get_logger(__name__)
+
+
+def list_groups(project=None, continuation_token=None, subject_types=None, organization=None, detect=None):
+    """ List all groups.
+    :param str project: List groups for a particular project.
+    :param str continuation_token : If there are more results than can't be returned in a single page, the result set
+                                    will contain a continuation token for retrieval of the next set of results.
+    :param [str] subject_types: A comma separated list of user subject subtypes to reduce the retrieved results.
+    """
+    organization = resolve_instance(detect=detect, organization=organization)
+    client = get_graph_client(organization)
+    scope_descriptor = None
+    if project is not None:
+        project = get_project_id_from_name(organization, project)
+        scope_descriptor = get_descriptor_from_storage_key(project, client)
+    if subject_types is not None:
+        subject_types = subject_types.split(',')
+    group_list_response = client.list_groups(scope_descriptor=scope_descriptor,
+                                             continuation_token=continuation_token, subject_types=subject_types)
+    if group_list_response.continuation_token is not None:
+        print('Showing only 500 groups. ' +
```

Comment:
This is not the best experience for automation. You removed the continuation token from the json response. 
It is ok to print it for table output but it should not be removed from json.

Reply:
Moved this to table formatting and retained the json response